[Alteqa] iP

Alteqa opened the pull request:
# a heading

- a
- bullet
- list

1. a 
2. numbered 
3. list
```ruby
a fenced code block (with syntax highlighting)
```
- [x] a 
- [ ] task 
- [ ] list

an emoji 🐒 
> a blockquote
[a hyperlink](https://pages.github.com/)
`inline code`
some text formatting: **bold**, _italic_, ~~strikethrough~~ etc.

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

Alteqa committed:
Changed bot name and programme

Alteqa committed:
added echo

Alteqa committed:
added add, list

Alteqa committed:
mark and unmark done

Alteqa committed:
added events with bugs and improved the checking answering system

Alteqa committed:
added todo, deadline, event

Alteqa committed:
added automation and fixed bugs

Alteqa committed:
updated automation

Alteqa committed:
added some exceptions

Alteqa committed:
added exceptions

Alteqa committed:
added delete

Alteqa committed:
added update to lists everytime it is modified

Alteqa committed:
save data to txt file

Alteqa committed:
Added recognition of date and time

Alteqa committed:
added storage class

Alteqa committed:
added ui class

Alteqa committed:
added parser

Alteqa committed:
added tasklist class

Alteqa committed:
Merge remote-tracking branch 'gradle_support/add-gradle-support' into branch-Level-8

Alteqa committed:
packaged all the classes

Alteqa committed:
added test classes

Alteqa committed:
Merge branch 'branch-Level-7'

Alteqa committed:
Merge branch 'branch-Level-8'

Alteqa committed:
added tasklist and storage tests

Alteqa committed:
test classes added

Alteqa committed:
added javadocs for 4 out of 7 of the classes

Alteqa committed:
made sure that the code follow conventions

Alteqa committed:
added search feature for words

Alteqa committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/hirwan/Parser.java
#	src/main/java/hirwan/hirwan.java

Alteqa committed:
fixed conflicts

Alteqa committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/hirwan/Parser.java
#	src/main/java/hirwan/hirwan.java

Alteqa committed:
fixed conflicts

Alteqa committed:
resolve some seach conflicts

Alteqa committed:
next if else

Alteqa committed:
resolved all conflicts

Alteqa committed:
adding GUI partially

Alteqa committed:
somemore gui additions with fxml

Alteqa committed:
added GUI

Alteqa committed:
working on GUI

wang-xinrong commented on src/main/java/hirwan/Hirwan.java:
```diff
@@ -0,0 +1,227 @@
+
+        Ui.output("Hello! " + logo);
+
+        Tasklist tasks = new Tasklist(Storage.read());
```
plural form used to indicate there are multiple tasks in the list, nice! 👍🏼

Alteqa replied:
Thanks!

wang-xinrong commented on src/main/java/hirwan/Hirwan.java:
```diff
@@ -0,0 +1,227 @@
+     * @return the list of indexes of tasks that contain the search word
+     */
+    public static List<Integer> searchWord(String word, List<String> tasks) {
+        List<Integer> indexes = new ArrayList<>();
```
similar to previous comment, nice use of plural form!

Alteqa replied:
Thanks again!

wang-xinrong commented on src/main/java/hirwan/Tasklist.java:
```diff
@@ -0,0 +1,40 @@
+            tasks.remove(index);
+    }
+
+    public int size() {
```
can probably use the name 'getSize' since it is a getter? verb would probably be better 😄

Alteqa replied:
Thats a good spot I agree with you

wang-xinrong commented on src/test/java/hirwan/StorageTest.java:
```diff
@@ -0,0 +1,43 @@
+
+        try {
+            File filePath = new File(FILE_PATH);
+            Scanner scan = new Scanner(filePath);
```
the noun 'scanner' would probably be a more suitable name than the verb 'scan' 😄

Alteqa replied:
isnt the scan of the file a noun?

wang-xinrong reviewed (commented):
Hirwan class is a bit long, may wish to pass some of its functionalities to other classes. Some minor modifications needed for some variable and method names too. But overall good effort in upholding OOP paradigm!

Alteqa committed:
added gui

Alteqa committed:
There is a lack of assert statements to detect bugs.

This makes the code more prone to major preventable errors that are caused by minor issues.

Added several assert statements to ensure that at certain points the code is behaving as desired.

Since assert statements require minimal resources to run and implement, they can be used plentifully.

Alteqa committed:
Merge pull request #1 from Alteqa/branch-A-Assertions

There is a lack of assert statements to detect bugs.

Alteqa committed:
created task class and classes for each task to abstract away similar functionality

Alteqa committed:
Code quality is improved from previous issues according to coding quality standard specified in the textbook

split the commands by classes due to the uneccesary length of the main method

this causes difficulty reading code and debugging

extracting the different commands into different classes enables us to properly look for the the methods that have errors or debug issues in modifications easily

Alteqa committed:
todo with errors

Alteqa committed:
update

Alteqa committed:
fixed event task

Alteqa committed:
Added period tasks and attempted to add undo

Alteqa committed:
finalised period tasks

Alteqa committed:
Update README.md

Alteqa committed:
Update README.md

Alteqa committed:
Update README.md

Alteqa committed:
updated gui

Alteqa committed:
Merge remote-tracking branch 'origin/master'

Alteqa committed:
Update README.md

Alteqa committed:
Update README.md

Alteqa committed:
changed picture

Alteqa committed:
Merge remote-tracking branch 'origin/master'

Alteqa committed:
added javadocs

Alteqa committed:
ui png

Alteqa committed:
Update README.md

Alteqa committed:
Update README.md

Alteqa committed:
Update README.md

Alteqa committed:
fixed path

Alteqa committed:
fixed bye command

Alteqa committed:
Merge remote-tracking branch 'origin/master'

Alteqa committed:
fixed jar file

Alteqa committed:
delete error

Alteqa committed:
Modify the Hirwan object name to hirwan instead of duke

Alteqa committed:
Remove sendButton variable that is never referenced

Alteqa committed:
Remove undo command

Alteqa committed:
Improve javadocs for Ui method

Alteqa committed:
Specify an UnmarkCommand method javadocs
